src/common/math.cairo

```diff
@@ -30,40 +31,36 @@ fn mul_inverse(x: felt252) -> felt252 {
 
 // Verifies that 0 <= x < RANGE_CHECK_BOUND
 fn assert_range_u128(x: felt252) {
-    let a: Option<u128> = x.try_into();
-    assert(a.is_some(), 'range check failed');
+    assert(TryInto::<felt252, u128>::try_into(x).is_some(), 'range check failed');
 }
 
 impl Felt252Div of Div<felt252> {
     fn div(lhs: felt252, rhs: felt252) -> felt252 {
-        let lhs_u256: u256 = lhs.into();
-        let rhs_u256: u256 = rhs.into();
-        (lhs_u256 / rhs_u256).try_into().unwrap()
+        (Into::<felt252, u256>::into(lhs) / Into::<felt252, u256>::into(rhs)).try_into().unwrap()
     }
 }
 
+fn div_rem2(lhs: felt252,) -> (felt252, felt252) {
+    let lhs: u256 = lhs.into();
+    let quotient = lhs / 2;
+    let remainder = lhs - (quotient * 2);
+    (quotient.try_into().unwrap(), remainder.try_into().unwrap())
+}
```

Comment:
```suggestion
fn div_rem2(lhs: felt252,) -> (felt252, felt252) {
    let (q, r) = DivRem::div_rem(Into::<felt252, u256>::into(lhs), 2_u256.try_into().unwrap());
    (q.try_into().unwrap(), r.try_into().unwrap())
}
```

Reply:
Please check if i didn't make some syntax mistake

Reply:
I decided to implement whole DivRem trait for `(felt252, felt252)`.